Makefile

```diff
@@ -0,0 +1,48 @@
+.PHONY: help
+
+#
+# Configuration variables
+#
+VENVDIR?=.venv
+VENV = $(VENVDIR)/bin
+
+PYTHON3OK :=$(shell python3 --version 2>&1)
+ifeq ($(PYTHON3OK),)
+  brew -y install python3.6 python3-pip
+endif
+
+help:
+	@grep -E '^[a-zA-Z_-]+:.*?## .*$$' $(MAKEFILE_LIST) | sort | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m%-30s\033[0m %s\n", $$1, $$2}'
```

Comment:
that "auto-documenting command" is a nifty trick. Maybe we should give credit to the author in a comment above it since this command is a 1-1 text match to the one found here:
https://marmelab.com/blog/2016/02/29/auto-documented-makefile.html
(linked on the Stackoverflow comment here https://stackoverflow.com/a/35730328/5601796)

Reply:
the text produced by `make help` has a big distance between the Make targets and their description. since we have short Make targets we could reduce the spacing from `30` to `15`
`%-30s` to `%-15s` in the `awk` `printf` format string

with `%-30s`:
```
$ make
report                         report kfp sample stats
test                           run kfp/tekton unit test
venv                           create and activate virtual environment
```

with `%-15s`:
```
$ make
help            Display the Make targets
report          Report compile status of the KFP testdata DSL scripts
test            Run compiler unit tests
venv            Create and activate virtual environment
```

Reply:
thanks , done.